src/Common/parseGlobs.cpp

```diff
@@ -48,6 +48,15 @@ std::string makeRegexpPatternFromGlobs(const std::string & initial_str_with_glob
 
         if (buffer.find(',') == std::string::npos)
         {
+            /// No dot or one dot in the filename. This is not a range.
+            if (buffer.find('.') == std::string::npos || buffer.find('.') == buffer.find_last_of('.'))
```

Comment:
It looks like an expression like `{file.csv.lz4}` will break this logic

Reply:
Indeed it will. Let me figure something out.

Reply:
Well obviously if we do something like `{file..csv}`, it will also break. Not sure how to improve that without rewriting this mess from scratch.

Reply:
I guess we can check that char to the left of first dot is a number, and char to the right of the last dot is a number. If dots don't follow each other, this is a filename. If there are more than two dots, this is a filename.

Reply:
Or just split regex in two and keep my sanity parsing a case of enum and range separately.